### PR TITLE
Try to play when receive wrong end time notification

### PR DESCRIPTION
**Pull request by raphrel**

### CODE REVIEW

File: Sources/Core/Services/PlaybackObservingService.swift [hunk 1]
```diff
@@ -65,14 +70,29 @@ final class ModernAVPlayerPlaybackObservingService: PlaybackObservingService {
                                                   object: nil)
     }
     
+    private func hasReallyReachedEndTime(player: AVPlayer) -> Bool {
+        guard
+            let duration = player.currentItem?.duration.seconds
+            else { return false }
+        
+        let currentTime = player.currentTime().seconds + 1
```
Comment (fmirault) [inline Sources/Core/Services/PlaybackObservingService.swift]: Maybe a comment to explain why you have to add 1 second ?

File: Tests/Services/ItemPlaybackObservingServiceTests.swift [hunk 2]
```diff
@@ -31,46 +31,82 @@ import Nimble
...
+            self.wasFailedToPlayToEndTime = false
+            self.tested = ModernAVPlayerPlaybackObservingService(player: self.player)
             self.tested.onPlaybackStalled = { [weak self] in self?.wasPlaybackStalledCalled = true }
             self.tested.onPlayToEndTime = { [weak self] in self?.wasPlayToEndTimeCalled = true }
+            self.tested.onFailedToPlayToEndTime = { [weak self] in self?.wasFailedToPlayToEndTime = true }
         }
         
-        context("init") {
-            it("should register to AVPlayerItemPlaybackStalled notification") {
+        context("trig AVPlayerItemPlaybackStalled notification") {
```
Comment (fmirault) [inline Tests/Services/ItemPlaybackObservingServiceTests.swift]: Replace trig with trigger

File: Tests/Services/ItemPlaybackObservingServiceTests.swift [hunk 3]
```diff
@@ -31,46 +31,82 @@ import Nimble
...
                 NotificationCenter.default.post(Notification.init(name: NSNotification.Name.AVPlayerItemPlaybackStalled))
                 
                 // ASSERT
                 expect(self.wasPlaybackStalledCalled).to(beTrue())
                 expect(self.wasPlayToEndTimeCalled).to(beFalse())
+                expect(self.wasFailedToPlayToEndTime).to(beFalse())
             }
+        }
+            
+        context("trig AVPlayerItemDidPlayToEndTime but item has reached his end time") {
```
Comment (fmirault) [inline Tests/Services/ItemPlaybackObservingServiceTests.swift]: Replace trig with trigger

File: Tests/Services/ItemPlaybackObservingServiceTests.swift [hunk 4]
```diff
@@ -31,46 +31,82 @@ import Nimble
...
                 NotificationCenter.default.post(Notification.init(name: NSNotification.Name.AVPlayerItemDidPlayToEndTime))
                 
                 // ASSERT
                 expect(self.wasPlaybackStalledCalled).to(beFalse())
                 expect(self.wasPlayToEndTimeCalled).to(beTrue())
+                expect(self.wasFailedToPlayToEndTime).to(beFalse())
             }
+        }
+        
+        context("trig AVPlayerItemDidPlayToEndTime but item has not reached his end time") {
```
Comment (fmirault) [inline Tests/Services/ItemPlaybackObservingServiceTests.swift]: Replace trig with trigger

File: Tests/Services/ItemPlaybackObservingServiceTests.swift [hunk 5]
```diff
@@ -31,46 +31,82 @@ import Nimble
...
+                
+                // ASSERT
+                expect(self.wasPlaybackStalledCalled).to(beFalse())
+                expect(self.wasPlayToEndTimeCalled).to(beFalse())
+                expect(self.wasFailedToPlayToEndTime).to(beTrue())
+            }
+        }
             
-            it("should register to AVPlayerItemFailedToPlayToEndTime notification") {
+        context("trig AVPlayerItemFailedToPlayToEndTime notification") {
```
Comment (fmirault) [inline Tests/Services/ItemPlaybackObservingServiceTests.swift]: Replace trig with trigger

File: Sources/Core/Services/PlaybackObservingService.swift (hunk 1 above)
Comment (raphrel) [reply Sources/Core/Services/PlaybackObservingService.swift]: yep

**Pull request merged**